DocumentService/Word/WordDocumentGenerator.cs

```diff
@@ -83,6 +86,9 @@ public static void GenerateDocumentByTemplate(string templateFilePath, DocumentD
                 // Write the document to output file path and close the document
                 WriteDocument(document, outputFilePath);
                 document.Close();
+
+                // Replace all the image placeholders in the output file
+                ReplaceImagePlaceholders(outputFilePath, outputFilePath, imagePlaceholders);
```

Comment:
Write an explanation why we are generating the document before replacing the images in the document. This will answer the question that anyone has when they go through the code.

Reply:
Added explanation for generating a document before replacing image and then opening it again for image method.

Reply:
Re-thrown the exception from public function.